src/test/bridges/tokemak/TokemakBridge.t.sol

```diff
@@ -0,0 +1,128 @@
+    address public constant T_WETH = 0xD3D13a578a53685B4ac36A1Bab31912D2B2A2F36;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant MANAGER = 0xA86e412109f77c45a3BC1c5870b880492Fb86A14;
+    address public constant DEPLOYER = 0x9e0bcE7ec474B481492610eB9dd5D69EB03718D5;
+
+    uint256 private nonce;
+    TokemakBridge private bridge;
+    AztecTypes.AztecAsset private empty;
+
+    uint256 private bridgeAddressId;
+
+    AztecTypes.AztecAsset private wAsset;
+    AztecTypes.AztecAsset private wtAsset;
+
+    event TokenBalance(uint256 previousBalance, uint256 newBalance);
+
+    function setUp() public {
+        bridge = new TokemakBridge(address(ROLLUP_PROCESSOR));
+        vm.deal(address(bridge), 0);
+        vm.startPrank(MULTI_SIG);
+
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 2000000);
+
+        ROLLUP_PROCESSOR.setSupportedAsset(WETH, 100000);
+
+        ROLLUP_PROCESSOR.setSupportedAsset(T_WETH, 100000);
+
+        vm.stopPrank();
+
+        bridgeAddressId = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+
+        wAsset = getRealAztecAsset(WETH);
+
+        wtAsset = getRealAztecAsset(T_WETH);
+    }
+
+    function testTokemakBridge() public {
+        validateTokemakBridge(1000, 500);
+    }
+
+    function validateTokemakBridge(uint256 _balance, uint256 _depositAmount) public {
+        deal(WETH, address(ROLLUP_PROCESSOR), _balance * 3);
+
+        //Deposit to Pool
+        uint256 output = depositToPool(_depositAmount);
+
+        //Request Withdraw
+        requestWithdrawFromPool(output);
+
+        //Next Cycle
+        uint256 newTimestamp = 1748641030;
+        vm.warp(newTimestamp);
+        vm.startPrank(DEPLOYER);
+        IManager(MANAGER).completeRollover("complete");
+        IManager(MANAGER).completeRollover("complete2");
+        vm.stopPrank();
+
+        //Test if automatic process withdrawal working
+        uint256 output2 = depositToPool(_depositAmount * 2);
+
+        nonce = getNextNonce();
+
+        //Request Withdraw
+        requestWithdrawFromPool(output2);
+
+        //Next Cycle
+        newTimestamp = 1758641030;
+        vm.warp(newTimestamp);
+        vm.startPrank(DEPLOYER);
+        IManager(MANAGER).completeRollover("complete3");
+        IManager(MANAGER).completeRollover("complete4");
+        vm.stopPrank();
+
+        //Withdraw
+        processPendingWithdrawal();
+    }
+
+    function depositToPool(uint256 _depositAmount) public returns (uint256) {
+        IERC20 assetToken = IERC20(WETH);
+        uint256 beforeBalance = assetToken.balanceOf(address(ROLLUP_PROCESSOR));
+
+        uint256 bridgeCallData = encodeBridgeCallData(bridgeAddressId, wAsset, empty, wtAsset, empty, 0);
+
+        (uint256 outputValueA, , ) = sendDefiRollup(bridgeCallData, _depositAmount);
+
+        uint256 afterBalance = assetToken.balanceOf(address(ROLLUP_PROCESSOR));
+
+        emit TokenBalance(beforeBalance, afterBalance);
```

Comment:
What is the purpose of having this event emitted here? For printouts during debugging you can use `emit log_uint()`. Also please don't spam the console log when the test passes.